ec2utils/ec2utilsbase/lib/ec2utils/ec2utils.py

```diff
@@ -39,6 +39,7 @@ def _connect(self):
             ec2 = boto3.client(
                 aws_access_key_id=self.access_key,
                 aws_secret_access_key=self.secret_key,
+                aws_session_token=self.session_token,
```

Comment:
I suspect this will break, the boto client is not as forgiving as one would hope with accepting arguments that are "None" and then doing the "right" thing.

So we either need to run a test before taking this code or change the code to be conditional, i.e.

if self.session_token:
  ec2 = boto3.client( aws_session_token=self.session_token,...)
else:
  ec2 = boto3.client(aws_access_key_id=self.access_key,...)

Reply:
I tried it locally on my machine at it worked fine!

Reply:
I changed it anyway to be sure ...